src/qx/Simulator.cpp

```diff
@@ -13,67 +13,59 @@
 
 namespace qx {
 
-std::optional<SimulationError> Simulator::set(std::string const &filePath) {
-    auto analyzer = cqasm::v1x::default_analyzer("1.2");
-
-    auto analysisResult = analyzer.analyze(filePath);
-    if (!analysisResult.errors.empty()) {
-        program.reset();
-        std::stringstream error;
-        error << "Cannot parse and analyze file " << filePath << ":";
-
-        std::for_each(analysisResult.errors.begin(), analysisResult.errors.end(), [&error](auto const& s) { error << "\n" << s; });
-
-        return SimulationError{error.str()};
-    }
+using AnalysisResult = cqasm::v1x::analyzer::AnalysisResult;
+using Program = cqasm::v1x::ast::One<cqasm::v1x::semantic::Program>;
 
-    program = analysisResult.root;
-    assert(!program.empty());
-
-    if (!program->error_model.empty() &&
-        program->error_model->name != "depolarizing_channel") {
-        program.reset();
-        return SimulationError{"Unknown error model: " + program->error_model->name};
-    }
+namespace {
+AnalysisResult parseCQASMV1File(std::string const &filePath) {
```

Comment:
Just a suggestion, I'm not the best at naming things: `parseCqasmV1xFile`. Just to avoid so many uppercase letters together. This is why I prefer snake case, which here would look like `parse_cqasm_v1_file`, but I understand the rest of the code is camel case.